src/pools/ID3MPool.sol

```diff
@@ -0,0 +1,120 @@
+
+    /**
+        @notice Withdraw assets (Dai) from the external pool.
+        @dev If the external pool requires a different amount to be passed in
+        the conversion should occur here as the Hub passes Dai [wad] amounts.
+        msg.sender must be authorized.
+        @param wad amount in asset (Dai) terms that we want to withdraw
+        @return bool whether the withdraw was successful
+    */
+    function withdraw(uint256 wad) external returns (bool);
+
+     /**
+        @notice Transfer shares.
+        @dev If the external pool/shares contract requires a different amount to be
+        passed in the conversion should occur here as the Hub passes Gem [wad]
+        amounts. msg.sender must be authorized.
+        @param dst address that should receive the shares
+        @param wad amount in Gem terms that we want to withdraw
+        @return bool whether the transfer was successful per ERC-20 standard
+    */
+    function transfer(address dst, uint256 wad) external returns (bool);
+
+    /**
+        @notice Transfer all shares from this pool.
+        @dev msg.sender must be authorized.
+        @param dst address that should receive the shares.
+        @return bool whether the transfer was successful per ERC-20 standard
+    */
+    function transferAll(address dst) external returns (bool);
+
+    /// @notice Some external pools require actions before debt changes
+    function preDebtChange() external;
+
+    /// @notice Some external pools require actions after debt changes
+    function postDebtChange() external;
+
+    /**
+        @notice Balance of assets this pool "owns".
+        @dev This could be greater than the amount the pool can withdraw due to
+        lack of liquidity.
+        @return uint256 number of assets in Dai [wad]
+    */
+    function assetBalance() external view returns (uint256);
+
+    /**
+        @notice Maximum number of assets the pool could deposit at present.
+        @return uint256 number of assets in Dai [wad]
+    */
+    function maxDeposit() external view returns (uint256);
+
+    /**
+        @notice Maximum number of assets the pool could withdraw at present.
+        @return uint256 number of assets in Dai [wad]
+    */
+    function maxWithdraw() external view returns (uint256);
+
+    /**
+        @notice Used to recover ERC-20 DAI accidentally sent to the pool.
+        --- YOU SHOULD NOT SEND ERC-20 DIRECTLY TO THIS CONTRACT ---
+        The presence of this function does not convey any right to recover tokens
+        sent to this contract. Maker Governance must evaluate and perform this
+        action at its sole discretion.
+        @dev msg.sender must be authorized.
+        @param dst address that should receive the shares
+        @param wad amount in wad terms that we want to withdraw
+        @return bool whether the transfer was successful per ERC-20 standard
+    */
+    function recoverDai(address dst, uint256 wad) external returns (bool);
```

Comment:
was this a group decision from the compound review? otherwise we were debating the need of this here: https://github.com/makerdao/dss-direct-deposit/pull/46#discussion_r878416594

Reply:
Yes, group decision, was discussed in length.

Reply:
gone f328965